cg/services/order_validation_service/rules/utils.py

```diff
@@ -28,12 +28,14 @@ def is_sample_on_plate(sample: Sample) -> bool:
 
 
 def is_application_not_compatible(
-    allowed_prep_categories: list[PrepCategory],
+    order_type: OrderType,
     application_tag: str,
     store: Store,
 ) -> bool:
-    application: Application = store.get_application_by_tag(application_tag)
-    return application and (application.prep_category not in allowed_prep_categories)
+    application: Application | None = store.get_application_by_tag(application_tag)
+    return application and order_type not in [
+        order_type_application.order_type for order_type_application in application.order_types
+    ]
 
```

Comment:
Clearer now. I do question the negation in the function though. Is it not clearer to ask `is_application_compatible` instead of the inverse? Ignore if this has been considered already

Reply:
Maybe having an `if not is_application_compatible` is clearer. I don't think there was much thought going into what direction, except for maybe having an `if not application: return True` might look odd in an `is_application_compatible` method. But I guess it is the same but in reverse now. I can change it

Reply:
You will always have a negation somewhere in this return statement - it will either be `return application and not ...` or `return not application or ...` depending on the chirality of the naming